src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,402 @@
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    soft_button[am::strings::type] = am::mobile_api::SoftButtonType::SBT_TEXT;
+    soft_button[am::strings::text] = "some text";
+    soft_button[am::strings::soft_button_id] = 15u;
+
+    smart_objects::SmartObject soft_buttons =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    soft_buttons[0] = soft_button;
+
+    msg_params[am::strings::soft_buttons] = soft_buttons;
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr CreateExpectedMessage() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    int32_t index = 0;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] =
+                  hmi_apis::Common_TextFieldName::navigationText1;
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kNavigationText1;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::navigationText2);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kNavigationText2;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] =
+                  static_cast<int32_t>(hmi_apis::Common_TextFieldName::ETA);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kEta;
+
+    msg_params
+        [am::hmi_request::navi_texts][index][am::hmi_request::field_name] =
+            static_cast<int32_t>(hmi_apis::Common_TextFieldName::totalDistance);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTotalDistance;
+
+    msg_params[am::hmi_request::navi_texts][index]
+              [am::hmi_request::field_name] = static_cast<int32_t>(
+                  hmi_apis::Common_TextFieldName::timeToDestination);
+    msg_params[am::hmi_request::navi_texts][index++]
+              [am::hmi_request::field_text] = kTimeToDestination;
+
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  void ExpectedManageMobileCommandResultCode(mobile_apis::Result::eType code) {
+    EXPECT_CALL(app_mngr_, ManageMobileCommand(MobileResultCodeIs(code), _))
+        .WillOnce(Return(true));
+  }
+
+ private:
+  void SetUp() OVERRIDE {
+    message_ = CreateMessageWithParams();
+    expected_message_ = CreateExpectedMessage();
+    mock_app_ = CreateMockApp();
+    ON_CALL(app_mngr_, application(kConnectionKey))
+        .WillByDefault(Return(mock_app_));
+    ON_CALL(app_mngr_, GetPolicyHandler())
+        .WillByDefault(ReturnRef(mock_policy_handler_interface_));
+  }
+
+ protected:
+  MessageSharedPtr message_;
```

Comment:
@BSolonenko These variables can be private

Reply:
@AByzhynar No, they can't. `TEST_F` function don't have access on `private` section.